apis/r/src/rinterface.cpp

```diff
@@ -72,14 +72,16 @@ Rcpp::List soma_array_reader(const std::string& uri,
         spdl::debug("[soma_array_reader] Selecting {} columns", column_names.size());
     }
 
+    auto tdb_result_order = get_tdb_result_order(result_order);
+
     // Read selected columns from the uri (return is unique_ptr<SOMAArray>)
-    auto sr = tdbs::SOMAArray::open(TILEDB_READ,
-                                    uri,
+    auto sr = tdbs::SOMAArray::open(OpenMode::read,
+				    uri,
                                     "unnamed",         // name parameter could be added
                                     platform_config,   // to add, done in iterated reader
                                     column_names,
                                     batch_size,
-                                    result_order);
+                                    tdb_result_order);
```

Comment:
Not the biggest of deals but if `result_order` needed protecting maybe sticking it into a namespace is an alternative?  `tdbs::` is used in other places.

But the prefix is fine too.